libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -207,9 +214,20 @@ queryCommitteeMembersState coldCredsFilter hotCredsFilter statusFilter nes = do
                 Nothing -> MemberNotAuthorized
                 Just Nothing -> MemberResigned
                 Just (Just hk) -> MemberAuthorized hk
-        -- TODO: implement after pulsing and snapshots are done
-        let nextEpochChange = NoChangeExpected
-        pure $ CommitteeMemberState hkStatus status mbExpiry nextEpochChange
+        pure $ CommitteeMemberState hkStatus status mbExpiry (nextEpochChange coldCred)
+
+      nextEpochChange :: Credential 'ColdCommitteeRole (EraCrypto era) -> NextEpochChange
+      nextEpochChange ck
+        | not inCurrent && inNext = ToBeEnacted
+        | not inNext = ToBeRemoved
+        | expiring = ToBeExpired
+        | otherwise = NoChangeExpected
+        where
+          lookupCurrent = Map.lookup ck comMembers
+          inCurrent = isJust lookupCurrent
+          inNext = Set.member ck nextComMembers
+          expiring = maybe False (< currentEpoch + 1) lookupCurrent
```

Comment:
I just realized something about this line. We are reporting that a member will expire by looking into current committee, but it could be possible that the member got his term extended and he will not expire. So we need to be looking into `nextComMembers` for this check.

One more important point is that ToBeExpired is a transition. So if member is already expired then there is no need to mention that he'll expire, it should be `NoChangeExpected`. With regards to the operator that checks expiration. If a member has the expiration set to the current epoch number that means in the next epoch that member will expire.

```suggestion
          expiring = maybe False (== currentEpoch) lookupFuture
```

Reply:
I think we need to look in both, current and future,  because a member can have ToBeExpired status even when there is no next commitee at all.

There is a case where the two, `ToBeExpired` and `TermAdjusted` are conflicting: 

For an existing committee member, we elect him again, with a Term that is : epoch-of-election + 1.
In this case, after the election, for this member both `TermAdjusted` or `ToBeExpired` kinda apply. 
I have made `ToBeExpired` to have priority - [so I'm only setting `TermAdjusted` in case the term was adjusted AND the new term is not expiring in the next epoch, otherwise it will be `ToBeExpired`](https://github.com/input-output-hk/cardano-ledger/blob/td/committee-query-next-epoch-change/libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs#L228)   Seems to be more relevant that hes about to expire, but I can see the argument for `TermAdjusted` too. If we go with `TermAdjusted` in this case, then the member would go from `TermAdjusted` directly to `Expired` + `NoChangeExpected`. 

I have a testcase for this in the other PR: https://github.com/input-output-hk/cardano-ledger/blob/td/committee-query-imp-test/libs/cardano-ledger-api/test/Test/Cardano/Ledger/Api/State/Imp/QuerySpec.hs#L186

Reply:
Your reasoning is 100% solid. I did think about this case as well after I made this suggestion. I was also leaning towards "ToBeExpired" being more relevant than "TermAdjusted". I was curious to see which was gonna be your preference :slightly_smiling_face: